core/src/components/item/item.tsx

```diff
@@ -359,7 +357,7 @@ export class Item implements ComponentInterface, AnchorInterface, ButtonInterfac
 
   private getFirstInteractive() {
     const controls = this.el.querySelectorAll<HTMLElement>(
-      'ion-toggle:not([disabled]), ion-checkbox:not([disabled]), ion-radio:not([disabled]), ion-select:not([disabled])'
+      'ion-toggle:not([disabled]), ion-checkbox:not([disabled]), ion-radio:not([disabled]), ion-select:not([disabled]), ion-input:not([disabled]), ion-textarea:not([disabled])'
```

Comment:
Nitpick: The padding around the input now has `cursor: pointer`, even though the rest of the control (label, input) doesn't; the inconsistency looks strange. For the other controls, it's consistent since `cursor: pointer` is used over the entire control.

Reply:
That `cursor: pointer` is coming from the `item-has-interactive-control` class on `ion-item`, and then it gets overwritten by `label`'s default styles. Either we could add `cursor: pointer` to the label --
```css
/* possibly restricted to just labels in inputs/textareas */
.item-has-interactive-control label {
  cursor: pointer;
}
```
-- or make items with inputs/textareas not get `cursor: pointer` at all, likely by having it not add the class in that case. I'm not really sure which is best though 🤔

Reply:
In https://github.com/ionic-team/ionic-framework/pull/29091/commits/143fb7d602f20be3a6e7d683e3c5710c01ce0afd I updated the code so that inputs and textures do not have the cursor effect. I updated the name of the class to be more descriptive of what it's doing, but I can change the name if you have other ideas.

Reply:
I could also do this with CSS and have it be `.item-has-interactive-control:not(:has(ion-input, ion-textarea))` but the specificity gets a little complicated, so I opted to just apply the class using logic in JS (since the class is already applied using JS logic)

Reply:
Maybe something like `item-has-click-toggle-control` or `item-control-needs-pointer-cursor`? "Needs cursor" by itself is confusing to me since in theory everything needs a cursor, like initially clicking into the input.

Reply:
Clarified in https://github.com/ionic-team/ionic-framework/pull/29091/commits/4362c3bb79fed288cbef47cfea03d316e12bde11 and https://github.com/ionic-team/ionic-framework/pull/29091/commits/0c9311c3af6de9ebd22bf4e26adcbd93db1a09a3